### PR TITLE
Add failing test for #17213

### PR DESCRIPTION
**Pull request by samcooke98**
I've tried to add a failing test for the issue described here. 
https://github.com/facebook/react/issues/17314

If you run it locally, and change `isMemo` to be false, the test will pass. The test will also pass if you change the line to 
`    const MemodSubbed = isMemo ? React.memo(Subbed, () => true) : Subbed;`

----

Below are my half-hearted attempts to look into this bug. 

It looks like when the `compare` function is defined it makes React call `createFiberFromTypeAndProps`. However, if the `compare` function is not defined, it means React goes down the codepath below; 

https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactFiberBeginWork.js#L380-L403

In other words, the bug is likely caused somewhere in `updateSimpleMemoComponent`

### CODE REVIEW

**Comment by gaearon** [comment]
Thank you for the test! Took us a while to get to this. Sorry this got autoclosed.
We think https://github.com/facebook/react/pull/18091 should fix it.

**Comment by samcooke98** [comment]
Thanks @gaearon ! Andrew told me as well: https://github.com/facebook/react/issues/17314#issuecomment-589446273 :D
